### PR TITLE
Implementing limit to actions for tracking

### PR DESCRIPTION
**Pull request by yashgiri**

### CODE REVIEW

File: packages/openshift/context.py [hunk 1]
```diff
@@ -493,17 +498,24 @@ def tracking(action_handler=None):
...
     managed Result object which can be accessed with get_result.
+    :param limit: If specified, it allows to specify a limit on the
+    number of actions stored by a given tracking context. If not specified
+    will store unlimited number of oc interactions.
     :return: The tracker contextmanager. If action_handler is not
     specified, call get_result to receive a Result object with all
     tracked Action objects.
     """
     c = Context()
+    if limit:
```
Comment (jupierce) [inline packages/openshift/context.py]: `0` in python is falsey, so consider what your developer would expect if they set limit=0 and what they would get instead.

File: packages/openshift/context.py [hunk 2]
```diff
@@ -493,17 +498,24 @@ def tracking(action_handler=None):
...
+        c.limit_tracking_actions = limit
```
Comment (jupierce) [inline packages/openshift/context.py]: Adding limit as a constructor argument of Result and passing it in for Result to enforce internally will probably simplify things here. It should also improve encapsulation (i.e. no need to expose `pop_first_action` to external callers).

File: packages/openshift/context.py [hunk 3]
```diff
@@ -319,7 +320,11 @@ def register_action(self, action):
 
             if c.tracking_strategy:
                 if isinstance(c.tracking_strategy, Result):
-                    c.tracking_strategy.add_action(action)
+                    if(len(c.tracking_strategy.actions())<c.limit_tracking_actions):
```
Comment (jupierce) [inline packages/openshift/context.py]: - For coding style, don't include outer parens for expressions that don't require it (see if statements that precede this one)
- I believe this logic is flawed when limit_tracking_actions is None.
Keep DRY when possible:
```
add_action(action)
if limit is not None and len(actions) > limit:
    pop
```

File: packages/openshift/context.py (hunk 1 above)
Comment (yashgiri) [reply packages/openshift/context.py]: The idea was to not let them keep it 0 but i see your point. Should i raise a warning for 0 or by default keep it a standard value ?

File: packages/openshift/result.py [hunk 4]
```diff
@@ -70,6 +75,14 @@ def as_json(self, indent=4, truncate_stdout=-1, redact_tokens=True, redact_strea
 
     def add_action(self, action):
         self.__actions.append(action)
+        if self.limit_tracking_actions is not None and self.actions_count() > self.limit_tracking_actions:
+            self.pop_first_action()
+
+    def actions_count(self):
+        return len(self.__actions)
+
+    def pop_first_action(self):
```
Comment (jupierce) [inline packages/openshift/result.py]: No need for these two methods (`actions_count` and `pop_first_action`) any more. It doesn't save code and doesn't expose anything we actually want the user of the object actually doing.

File: packages/openshift/result.py [hunk 5]
```diff
@@ -3,9 +3,14 @@
 
 
 class Result(object):
-    def __init__(self, high_level_operation):
+    def __init__(self, high_level_operation, track_limit = None):
```
Comment (jupierce) [inline packages/openshift/result.py]: For consistency, lets call this `tracking_limit` vs `track_limit`.

Comment (yashgiri) [reply packages/openshift/result.py]: got it

**Pull request merged**